cli/tools/upgrade.rs

```diff
@@ -41,6 +41,16 @@ pub async fn upgrade_command(
 
   let install_version = match version {
     Some(passed_version) => {
+      if canary
+        && !regex::Regex::new("^[0-9a-f]{40}$")?.is_match(&passed_version)
```

Comment:
Should this support upper case hashes too? (Add `A-F`? Or is it already case insensitive?)

Reply:
Pretty sure git hashes are just lowercase